### PR TITLE
Fix tests on 41 branch and temporarily disable those unfixable

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: datafusion/core/tests/fuzz_cases/window_fuzz.rs [hunk 1]
```diff
@@ -146,6 +163,7 @@ async fn window_bounded_window_random_comparison() -> Result<()> {
 // This tests whether we can generate bounded window results for each input
 // batch immediately for causal window frames.
 #[tokio::test(flavor = "multi_thread", worker_threads = 16)]
+#[ignore = "Non-nullable window count(*) can currently return NULL value in some edge cases. See tmp_test()"]
```
Comment (milevin) [inline datafusion/core/tests/fuzz_cases/window_fuzz.rs:166]: Only in our branch? What's special about it?

File: datafusion/sqllogictest/test_files/window.slt [hunk 2]
```diff
@@ -833,19 +833,13 @@ LIMIT 5
 
 
 #fn window_frame_ranges_preceding_and_preceding
-query II
+query error DataFusion error: Arrow error: Invalid argument error: Column 'count\(\*\) ORDER BY \[aggregate_test_100\.c2 ASC NULLS LAST\] RANGE BETWEEN 3 PRECEDING AND 1 PRECEDING' is declared as non\-nullable but contains null values
```
Comment (milevin) [inline datafusion/sqllogictest/test_files/window.slt:836]: Is it the same as the ignored test above? If so, should this be ignored too?

**Review by milevin — APPROVED** [review]

File: datafusion/core/tests/fuzz_cases/window_fuzz.rs [hunk 3]
```diff
@@ -50,6 +50,23 @@ use rand::rngs::StdRng;
...
+    let ctx = SessionContext::new_with_config(session_config);
+
+    let result = ctx.sql("SELECT x, count(x) OVER (ORDER BY x NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING) AS c FROM (VALUES (1), (2), (NULL), (3)) t(x)").await?.collect().await;
+    assert!(result.is_err());
+    // When this fails, drop the commit that introduce this test. This will unignore temporarily disabled tests.
+    result
+        .err()
+        .unwrap()
+        .message()
+        .contains("is declared as non-nullable but contains null values");
```
Comment (vgapeyev) [inline datafusion/core/tests/fuzz_cases/window_fuzz.rs]: Should there be another `assert!` around this `result....contains()` check?

File: datafusion/sqllogictest/test_files/window.slt (hunk 2 above)
Comment (vgapeyev) [reply datafusion/sqllogictest/test_files/window.slt:836]: There is no 'ignore' in SLT, so, as I understand, this one is effectively ignored by expecting the error?

**Review by vgapeyev — APPROVED** [review]
Looks reasonable -- modulo the possibly-missing assert. Or did I misunderstand what's going on there?

**Comment by vgapeyev** [comment]
Also, maybe rename the PR to "Temporarily disable failing tests on 41 branch"?

File: datafusion/core/tests/fuzz_cases/window_fuzz.rs (hunk 1 above)
Comment (findepi) [reply datafusion/core/tests/fuzz_cases/window_fuzz.rs:166]: So on our fork we have a commit:

- "Physical-plan counterpart for DF https://github.com/apache/datafusion/pull/11256" (https://github.com/sdf-labs/arrow-datafusion/commit/d595069bf95db95ad14ce1826d482aee13a526e6)

It breaks tests and uncovers bugs. Fixing most of the test is easy (just update expected value), but there are 2+ problematic ones:  `bounded_window_causal_non_causal`, `window_bounded_window_random_comparison`. They fail (`Column 'count' is declared as non-nullable but contains null values`) because the schema returned marks count as null, but implementation (incorrectly) returns null for queries like `SELECT count(x) OVER (ORDER BY x NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING) AS c FROM (VALUES (1), (2), (NULL), (3)) t(x)`

- In DF upstream this was fixed in https://github.com/apache/datafusion/pull/11989  but it's not easily cherry-pickable (i tried)
- also,  https://github.com/apache/datafusion/issues/13065 suggests it's potentially disruptive change

Comment (findepi) [reply datafusion/core/tests/fuzz_cases/window_fuzz.rs:166]: this is a TMP commit we will drop during next update. there is a test making sure we don't forget, even if we don't read commit titles

File: datafusion/sqllogictest/test_files/window.slt (hunk 2 above)
Comment (findepi) [reply datafusion/sqllogictest/test_files/window.slt:836]: it's the same commit as ignored tests. there is a test that -- when starting to fail (ie during update) -- will ask for the whole commit to be reverted, and then this gets reverted as well

File: datafusion/core/tests/fuzz_cases/window_fuzz.rs (hunk 3 above)
Comment (findepi) [reply datafusion/core/tests/fuzz_cases/window_fuzz.rs]: keen eye. thanks for catching me fall

**Comment by findepi** [comment]
> Also, maybe rename the PR to "Temporarily disable failing tests on 41 branch"?

Most of tests is fixed, not just disabled.
Changed to "Fix tests on 41 branch and temporarily disable those unfixable"

File: datafusion/core/tests/fuzz_cases/window_fuzz.rs (hunk 1 above)
Comment (vgapeyev) [reply datafusion/core/tests/fuzz_cases/window_fuzz.rs:166]: @findepi, I bow in awe for your catching, understanding and illuminating the consequences of that seemingy-innocent count() nullability tweak! (in DF 39, if I recall?) 
My fix in https://github.com/sdf-labs/arrow-datafusion/commit/d595069bf95db95ad14ce1826d482aee13a526e6 was sheer luck -- both in finding it and not in triggering more problems, which you uncovered were lurking there!

Comment (findepi) [reply datafusion/core/tests/fuzz_cases/window_fuzz.rs:166]: @vgapeyev i just followed failing test cases, nothing special.
Then i realized the upstream has equivalent of your fix, so i `git bisect`-ed to find the commit which did it (grepping `git log` for "count" or "window" couldn't find the right commit).
